pkg/extension/create_interceptor.go

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package extension
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/storage"
+)
+
+type namedCreateAPIFunc struct {
+	Name string
+	Func func(InterceptCreateOnAPI) InterceptCreateOnAPI
+}
+
+type namedCreateTxFunc struct {
+	Name string
+	Func func(InterceptCreateOnTx) InterceptCreateOnTx
+}
+
+type createHookOnAPIHandler struct {
+	CreateHookOnAPIFuncs         []*namedCreateAPIFunc
```

Comment:
change to something like

```
type createHookOnAPIHandler struct {
	CreateHookOnAPIFuncs         map[string]func(InterceptCreateOnAPI) InterceptCreateOnAPI
	CreateHookOnTxFuncs         map[string]func(InterceptCreateOnTx) InterceptCreateOnTx
}
```

Reply:
I tried this, but map does not have iteration order so we cannot order the functions